go/vt/vttablet/tabletserver/vstreamer/planbuilder.go

```diff
@@ -0,0 +1,361 @@
+/*
+Copyright 2018 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vstreamer
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+	"time"
+
+	"vitess.io/vitess/go/mysql"
+	"vitess.io/vitess/go/sqltypes"
+	"vitess.io/vitess/go/vt/key"
+	"vitess.io/vitess/go/vt/sqlparser"
+	"vitess.io/vitess/go/vt/vtgate/vindexes"
+	"vitess.io/vitess/go/vt/vttablet/tabletserver/schema"
+
+	binlogdatapb "vitess.io/vitess/go/vt/proto/binlogdata"
+	querypb "vitess.io/vitess/go/vt/proto/query"
+	topodatapb "vitess.io/vitess/go/vt/proto/topodata"
+)
+
+// Plan represents the streaming plan for a table.
+type Plan struct {
+	Table        *Table
+	ColExprs     []ColExpr
+	VindexColumn int
+	Vindex       vindexes.Vindex
+	KeyRange     *topodatapb.KeyRange
+}
+
+// ColExpr represents a column expression.
+type ColExpr struct {
+	ColNum    int
+	Alias     sqlparser.ColIdent
+	Type      querypb.Type
+	Operation Operation
+}
+
+// Operation represents the operation to be performed on a column.
+type Operation int
+
+// The following are the supported operations on a column.
+const (
+	OpNone = Operation(iota)
+	OpMonth
+	OpDay
+	OpHour
+)
+
+// Table contains the metadata for a table. The
+// name is dervied from mysql's Table_map_log_event.
+type Table struct {
+	*mysql.TableMap
+	Columns []schema.TableColumn
+}
+
+func (plan *Plan) filter(values []sqltypes.Value) (bool, []sqltypes.Value, error) {
+	result := make([]sqltypes.Value, len(plan.ColExprs))
+	for i, colExpr := range plan.ColExprs {
+		switch colExpr.Operation {
+		case OpMonth:
+			v, _ := sqltypes.ToInt64(values[colExpr.ColNum])
```

Comment:
Per discussion in Slack, I think it will be good to add a comment on why this processing is happening in this side and not defer to the player.